lib/modules/platform/azure/index.ts

```diff
@@ -990,3 +994,42 @@ export async function deleteLabel(
   const azureApiGit = await azureApi.gitApi();
   await azureApiGit.deletePullRequestLabels(config.repoId, prNumber, label);
 }
+
+function cachePr(pr?: GitPullRequest): void {
+  config.prList ??= [];
+  // istanbul ignore if
+  if (!pr) {
+    return;
+  }
+
+  const azurePr = getRenovatePRFormat(pr);
+  const idx = getCachedPrIdx(azurePr.number);
+  if (is.undefined(idx)) {
+    config.prList.push(azurePr);
+  } else {
+    config.prList[idx] = azurePr;
+  }
+}
+
+function updateCachedPr(prNo: number, pr?: GitPullRequest): void {
+  config.prList ??= [];
+  const idx = getCachedPrIdx(prNo);
+  if (!pr || is.undefined(idx)) {
+    // if pr doesn't exist in cache then invalidate cache so we get new list from api next time.
+    config.prList = undefined as any;
+    return;
+  }
+
+  const cachedPr = config.prList[idx];
+  config.prList[idx] = getRenovatePrFormatForCachedPr(cachedPr, pr);
```

Comment:
```suggestion
  getRenovatePrFormatForCachedPr(cachedPr, pr);
```

This function already mutates `cachedPr`

Reply:
hmm.. im not sending the list though. im sending 
const cachedPr = config.prList[idx];
if i take it out into a new const does it get the same address?
this doesn't happen in java. it just creates a new address for the cachedPr.